### PR TITLE
Add telemetry client events handler

### PR DESCRIPTION
**Pull request by larkox**
#### Summary
Add telemetry client events handler

#### Ticket Link
None

### CODE REVIEW

File: experimental/telemetry/tracker.go [hunk 1]
```diff
@@ -105,3 +108,28 @@ func (t *tracker) TrackUserEvent(event, userID string, properties map[string]int
...
+func (t *tracker) HandleClientEvent(w http.ResponseWriter, r *http.Request) {
+	userID := r.Header.Get("Mattermost-User-ID")
+	if userID == "" {
+		http.Error(w, "Not authorized", http.StatusUnauthorized)
+		return
+	}
+
+	var telemetryRequest *telemetryAPIRequest
+	decoder := json.NewDecoder(r.Body)
+	err := decoder.Decode(&telemetryRequest)
```
Comment (hanzei) [inline experimental/telemetry/tracker.go:126]: Nit:
```suggestion
	err := json.NewDecoder(r.Body).Decode(&telemetryRequest)
```

File: experimental/telemetry/tracker.go [hunk 2]
```diff
@@ -105,3 +108,28 @@ func (t *tracker) TrackUserEvent(event, userID string, properties map[string]int
...
+	if err != nil {
+		http.Error(w, "Unable to decode JSON", http.StatusBadRequest)
```
Comment (hanzei) [inline experimental/telemetry/tracker.go:128]: It's a bummer that there is no entity to log the error

File: experimental/telemetry/tracker.go [hunk 3]
```diff
@@ -105,3 +108,28 @@ func (t *tracker) TrackUserEvent(event, userID string, properties map[string]int
...
+		return
+	}
+
+	if telemetryRequest.Event != "" {
+		_ = t.TrackUserEvent(telemetryRequest.Event, userID, telemetryRequest.Properties)
```
Comment (hanzei) [inline experimental/telemetry/tracker.go:133]: Same here

**Review by hanzei — APPROVED** [review]
LGTM :+1: 

I would love to log the errors, but that is out of scope of this PR

File: experimental/telemetry/tracker.go (hunk 2 above)
Comment (larkox) [reply experimental/telemetry/tracker.go:128]: What we can do is to return the error, and leave the http handling to the caller. What do you think?

Comment (hanzei) [reply experimental/telemetry/tracker.go:128]: Leaving it to the caller would push the error to the webapp in most of the cases. For debugging it's more useful to log the error in the server logs.

Comment (larkox) [reply experimental/telemetry/tracker.go]: Right now the tracker does not have "logging functions", and I would prefer to leave it like that. I made some changes to return the error.

File: experimental/telemetry/tracker.go [hunk 4]
```diff
@@ -105,3 +107,30 @@ func (t *tracker) TrackUserEvent(event, userID string, properties map[string]int
 	properties["UserActualID"] = userID
 	return t.TrackEvent(event, properties)
 }
+
+type telemetryAPIRequest struct {
+	Event      string
+	Properties map[string]interface{}
+}
+
+func (t *tracker) HandleClientEvent(r *http.Request) error {
```
Comment (hanzei) [inline experimental/telemetry/tracker.go:116]: I'm not sure if that signature is better. With `(w http.ResponseWriter, r *http.Request)` is was easily plugable and with that change a consumer as to write there own error handling method. I would like to keep the methods a just plug an play.

**Pull request closed without merging**